core/src/main/java/org/mule/config/ExceptionHelper.java

```diff
@@ -541,34 +543,63 @@ public static Map getExceptionInfo(Throwable t)
 
     public static String getExceptionStack(Throwable t)
     {
+        Throwable root = getRootException(t);
+        MuleException rootMule = getRootMuleException(t);
+
         StringBuilder buf = new StringBuilder();
-        // get exception stack
-        List<Throwable> exceptions = getExceptionsAsList(t);
 
-        int i = 1;
-        for (Iterator<Throwable> iterator = exceptions.iterator(); iterator.hasNext(); i++)
+        ExceptionReader er = getExceptionReader(rootMule);
+        buf.append(er.getMessage(rootMule)).append(" (")
+            .append(rootMule.getClass().getName())
+            .append(")")
+            .append(SystemUtils.LINE_SEPARATOR);
+
+        if (verbose)
         {
-            if (i > exceptionThreshold && exceptionThreshold > 0)
+            int processedElements = 0;
+            int nonMuleElements = 0;
+            int processedMuleElements = 1;
+            for (StackTraceElement stackTraceElement : root.getStackTrace())
             {
-                buf.append("(").append(exceptions.size() - i + 1).append(" more...)");
-                break;
+                if (processedMuleElements > exceptionThreshold && exceptionThreshold > 0)
```

Comment:
exceptionThreshold > 0 is redundant here. If you want to be sure of that, then check it on the class initialization